### PR TITLE
Blog app: add devise

### PR DESCRIPTION
**Pull request by AliShahSafdari**
In this PR we did:

- Remove current_user method in ApplicationController, devise will provide us with one.

- Install and setup devise.
      -Can register a new user.
       -User logs in with a combination of email and password.
       - Hashed password should be stored in the database.
       -Ask for confirmation of email.
        -Can reset password.

- Modify the views in devise for registration and login to match the wireframes from

### CODE REVIEW

**Comment by AliShahSafdari** [comment]
Sorry 🙏, I did  a mistake I requested review in Day 4 It was the project of day 2

**Review by Davidosky007 — COMMENTED** [review]
Hi @AliShahSafdari @Salimer 👋, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted the wrong project for `Blog app: add API endpoints` kindly submit the right project 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: app/models/user.rb [hunk 1]
```diff
@@ -1,13 +1,20 @@
...
   has_many :posts, foreign_key: 'author_id'
   has_many :comments, foreign_key: 'author_id'
   has_many :likes, foreign_key: 'author_id'
 
+  # Attributes and Validations
   validates :name, presence: true, length: { in: 3..25 }
-  validates :photo, presence: true
-  validates :bio, presence: true, length: { in: 10..255 }
+  # validates :photo, presence: true
+  # validates :bio, presence: true, length: { in: 10..255 }
```
Comment (danmainah) [inline app/models/user.rb]: - [ ] @AliShahSafdari and @Salimer Great work adding devise in your project, however its not good practise having commented code in your codebase, kindly consider removing the commented code.

File: app/views/devise/registrations/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+        <div class="form-group">
+          <%= f.label :password_confirmation %>
+          <%= f.password_field :password_confirmation, autocomplete: "new-password", class: "form-control" %>
+        </div>
+
+        <div class="text-center">
+          <%= f.submit "Sign up", class: "btn btn-primary" %>
+        </div>
+      <% end %>
```
Comment (danmainah) [inline app/views/devise/registrations/new.html.erb]: - [ ] @Team Great work on your registration page, however in order for your layout to match the design given kindly ensure the width of your button is the same size as the input fields size, check the comparison below for a better understanding:

Given Design          |  Your design     
:-------------------------:|:-------------------------:
image  |  image

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @AliShahSafdari and @Salimer , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ Added devise and ensured the user can successfully sign up and sign in.
✔️ Ensure the user needs to authenticate the signup by confirming their email.


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: app/models/user.rb (hunk 1 above)
Comment (AliShahSafdari) [reply app/models/user.rb]: Thanks for the good advice.

**Review by codepantha — APPROVED** [review]
# STATUS: APPROVED 🟢 ✔️ 

## Hi team👋, 

Great job implementing the required changes 👏 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights 🎯 
✔️ Descriptive PR
✔️ Implemented authentication with devise 📌 
✔️ Registration and login views match the wireframes and their styling 📌 
✔️ No linter errors 🟢 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**